docs/user_guide/datasets.rst

```diff
@@ -0,0 +1,394 @@
+Datasets
```

Comment:
I feel like we should provide a full list of references used for this as well. I'm sure several of Vincent's blogs/videos/etc. provided the foundation for this. A few are mentioned in #507 I think. If there's no direct reference we can always acknowledge these resources as inspiration/foundation and link to them for people to check out. Wdyt?

Reply:
Yeah, would that go under the References section? How would we cite them, in the same way as the papers?